README.rst

```diff
@@ -103,24 +103,161 @@ Available Features
+
+Fork and clone the forked repository:
+
+.. code-block:: shell
+
+    git clone git://github.com/<your_fork>/openwisp-monitoring
+
+Navigate into the cloned repository:
+
+.. code-block:: shell
+
+    cd openwisp-monitoring/
+
+Start Redis and InfluxDB using Docker:
+
+.. code-block:: shell
 
-We use InfluxDB to store metrics and Redis as celery broker (you can use a different
-broker if you want). The recommended way for development is running them using Docker
-so you will need to `install docker and docker-compose <https://docs.docker.com/engine/install/>`_
-beforehand.
+    docker-compose up -d redis influxdb
+
+Setup and activate a virtual-environment. (we'll be using  `virtualenv <https://pypi.org/project/virtualenv/>`_)
+
+.. code-block:: shell
+
+    python -m virtualenv env
+    source env/bin/activate
+
+Make sure that you are using pip version 20.2.4 before moving to the next step:
+
+.. code-block:: shell
+
+    pip install -U pip wheel setuptools
+
+Install development dependencies:
+
+.. code-block:: shell
+
+    pip install -e .
+    pip install -r requirements-test.txt
+    npm install -g jshint stylelint
+
+Install WebDriver for Chromium for your browser version from `<https://chromedriver.chromium.org/home>`_
+and extract ``chromedriver`` to one of directories from your ``$PATH`` (example: ``~/.local/bin/``).
+
+Create database:
+
+.. code-block:: shell
+
+    cd tests/
+    ./manage.py migrate
+    ./manage.py createsuperuser
 
-In case you prefer not to use Docker you can `install InfluxDB <https://docs.influxdata.com/influxdb/v1.8/introduction/install/>`_
-and Redis from your repositories, but keep in mind that the version packaged by your distribution may be different.
+Run celery and celery-beat with the following commands (separate terminal windows are needed):
 
-Install spatialite and sqlite:
+.. code-block:: shell
+
+    cd tests/
+    celery -A openwisp2 worker -l info
+    celery -A openwisp2 beat -l info
+
+Launch development server:
 
 .. code-block:: shell
 
-    sudo apt-get install -y sqlite3 libsqlite3-dev openssl libssl-dev \
-                            gdal-bin libproj-dev libgeos-dev libspatialite-dev \
-                            fping
+    ./manage.py runserver 0.0.0.0:8000
+
+You can access the admin interface at http://127.0.0.1:8000/admin/.
+
+Run tests with:
+
+.. code-block:: shell
+
+    ./runtests.py --parallel
+
+Run quality assurance tests with:
+
+.. code-block:: shell
+
+    ./run-qa-checks
+
+Install and run on docker
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Note**: This Docker image is for development purposes only.
+For the official OpenWISP Docker images, see: `docker-openwisp
+<https://github.com/openwisp/docker-openwisp>`_.
+
+Build from the Dockerfile:
+
+.. code-block:: shell
+
+    docker-compose build
+
+Run the docker container:
+
+.. code-block:: shell
+
+    docker-compose up
 
 Setup (integrate in an existing Django project)
 -----------------------------------------------
```

Comment:
I think we should move this section under the installation instructions as we have in OpenWISP Controller